### ISSUE TITLE
[aws-ecr-assets] DockerImageAsset: Suppress docker build output

### ISSUE THREAD

**Issue by nikolic-milan** [aws/aws-cdk#29555]
### Describe the feature

I am using `ecr_assets.DockerImageAsset` to build a docker image from a Dockerfile. I would like a way to suppress the output of this. The output from docker interferes with the normal out of CDK, below example of how it looks. Ugly and more importantly confusing. 
![image](https://github.com/aws/aws-cdk/assets/39311195/1ccc2d06-434f-4288-9103-3d1e58501c43)

It would be nice, if either one could disable it, our somehow guarantee that they don't interfere, build the image and then once done and all the logs are out, start with the cdk output.

### Use Case

I just think it would be better if one could suppress the output and just have the regular CDK output. It's going to be easier when debugging, and if I need to debug the build image process I could temporarily remove the flag.

### Proposed Solution

`docker build` has a flag that suppresses outputs, `-q --quite`. A solution similar to how `cache_disabled` is implemented. 
An optional `boolean` parameter that adds `-q` flag to the `docker build` command. 

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.130.0

### Environment details (OS name and version, etc.)

Windows 11

**Comment by pahud** [comment]
Makes sense for me. Adding this as a p2 feature request and we welcome the pull requests.

**Comment by kaleko** [comment]
Any update on this feature? My users are confused during `cdk deploy` because the verbose docker output pushes the `y/n` user input request out of the window, so they think the deployment is hanging.

**Comment by zencircle** [comment]
`BUILDKIT_PROGRESS=quiet cdk diff myStack 2>&1 | grep -Ev "^Sending build|^Step [0-9]+|^--->|^DEPRECATED|^BuildKit|^ ---> |^Successfully built|^Successfully tagged"` - Word around till fix is implemented
